### DISCUSSION TITLE
concepts/chain/

### DISCUSSION THREAD

**Discussion by giscus** [go-gost/gost#470] [General]
# concepts/chain/

https://gost.run/concepts/chain/

**Comment by KelvinF97** [comment]
请问基于单点主机转发多个节点的隧道代理模式，使用转发链组（chainGroup）合适还是使用节点组（nodes）合适？求大佬指点，主要是负载均衡，做到单个代理节点转发到后面多个节点。

**Comment by wiki4j** [comment]
转发链组能否支持分流。我需要根据不同域名选择不同的转发链组，才能在单个代理端口上聚合各种奇葩的网络路点。之前是用tinyproxy+多个gost来做聚合，现在是尝试使用gost单进程达成。
跳跃点和转发节点上的分流，虽然提供了一定程度上的路由规则，但是：
1、多个分流如果使用白名单，组合的结果不是期望（貌似逻辑是不被任何一个分流器阻拦）
2、我用hops和nodes两层虽然也大体完成了按域名路由，但是写起来太费劲了，没点强迫症直接就放弃了

**Reply by zero-custom** [reply]
关于你的问题，issues#657有回答，我的总结就是：bypass+whitelist只是让网站可选bypass节点和无规则节点
我现在的做法：
最早：1个chain，1个hop，N个带bypass+whitelist的node，不配置无bypass的节点，这样就能保证白名单的网站能匹配节点，不需要过节点的能用保底回落本地出口
现在：基本一致，但是在bypass上用插件替代gost的原生白名单，还做了自动规则更新
未来：准备在hop上用插件实现基于白名单的节点返回，预计实现自动读节点(✓)、自动读规则(✓)、自动的规则组合

**Reply by wiki4j** [reply]
"不配置无bypass的节点"  问题就在于我就是想有规则的走规则分流走，剩下的就走无规则的那个（比如默认走visual直连）。我现在为了达成这个效果，把所有其他规则的黑白名单整到一个新的规则，然后取反。可废脑子了
